src/themes/gradients.ts

```diff
@@ -1,9 +1,15 @@
+import { Appearance } from 'react-native'
+
+import Colors from './colors'
+
 /**
  * The gradients are very specific to the consumption of react-native-linear-gradient.
  */
 export const gradients = {
   TransparentToBlack: {
-    colors: [ 'rgba(0, 0, 0, 0)', 'rgba(0, 0, 0, 1)' ],
+    colors: Appearance.getColorScheme() === 'light'
+      ? [ 'rgba(255,255,255,0)', Colors.MainView ]
+      : [ 'rgba(0,0,0,0)', Colors.MainView ],
```

Comment:
You may just want to name this `TransparentToMain` and use a similar approach with the objects as in `colors.ts` with `const gradients = Appearance.getColorScheme() === 'light' ? lightGradients : darkGradients`, since I suspect every gradient will be subject to the theme applied. 

Alternatively, if you do not see this being different, I'd suggest a refactor to the following to improve clarity and reduce verbosity:


```
const isDark = Appearance.getColorScheme() === 'dark'

export const gradients = {
  TransparentToBlack: {
    colors: [ isDark ? 'rgba(0, 0, 0, 0)' : 'rgba(255, 255, 255, 0), Colors.MainView ],
    locations: [ 0.0, 0.6 ],
  },
}
```

Although, surely it wouldn't matter what `rgba(x, y, z, 0)` are when the alpha channel is `0`?

Reply:
There is a typing issue - using `OpaqueColor` is not supported - it's a good thing TS flagged this. The library does not support using `PlatformColor` on android as a value to it - there is a PR open there that adds support. We may either use that PR in our codebase instead, or use a fixed set of colour values here...

Reply:
Actually that is the main issue here, when you do rgba(0, 0, 0, 0) and go to rgba (255, 255, 255, 1) you are actually going from a transparent black to a solid white. There is a gradient of black to white still :(

Reply:
I tried very hard to somehow extract the rgb values from the Colors.MainView to use with a custom alpha, but without success. The PlatformColor is using a semantic value, not an actual color.